File: datasets/IR-Plag/convertai-original-java-rest-kotlin/case-07/plagiarized/L6/01/L6.kt
package IR

import java.util.*

fun jumlahDiagonal(m: Array<DoubleArray>): Double {
    var sum = 0.0
    var hit = m.size - 1

    while (hit >= 0) {
        sum += m[hit][hit]
        hit--
    }
    return sum
}

fun print(m: Array<DoubleArray>) {
    println("Sum of the elements in the major diagonal is $jumlahDiagonal(m)")
}

fun main() {
    val sc = Scanner(System.`in`)
    val m = Array(4) { DoubleArray(4) }
    println("Enter a 4 by 4 matrix row by row: ")

    for (i in 3 downTo 0) {
        for (j in 3 downTo 0) {
            m[i][j] = sc.nextDouble()
        }
    }
    print(m)
}

File: datasets/IR-Plag/convertai-original-java-rest-typescript/case-07/plagiarized/L6/01/L6.ts
import { Scanner } from 'readline-sync';

function jumlahDiagonal(m: number[][]): number {
    let sum = 0;
    let hit = m.length - 1;

    while (hit >= 0) {
        sum += m[hit][hit];
        hit--;
    }
    return sum;
}

function print(m: number[][]): void {
    console.log(`Sum of the elements in the major diagonal is ${jumlahDiagonal(m)}`);
}

function main(): void {
    const sc = new Scanner();
    const m: number[][] = new Array(4).fill(0).map(() => new Array(4).fill(0));
    console.log('Enter a 4 by 4 matrix row by row: ');

    for (let i = 3; i >= 0; i--) {
        for (let j = 3; j >= 0; j--) {
            m[i][j] = sc.nextDouble();
        }
    }
    print(m);
}

main();

